### DISCUSSION TITLE
How to remove the contents of the images?

### DISCUSSION THREAD

**Discussion by yusheng0104** [pymupdf/PyMuPDF#1346] [Q&A] [answered]
Hi,

Is there a method could remove all the contents of an image in a pdf file? Pymupdf is sometimes too powerful and picks up some words from images. And my goal is analyzing the text not in the image. 

Thanks

**Comment by JorjMcKie** [comment, marked as answer]
I think I don't understand your problem.
There are several ways how to ignore images and only extract text: page method ``get_text`` has an option string as argument one, and a keyword parameter `flags`, which together control this:
* option = "text" will only extract text
* switching off ``TEXT_PRESERVE_IMAGES`` from ``flags`` will only extract text for the other option values ("blocjs", "dict" etc.)

Please be more specific.

**Reply by yusheng0104** [reply]
Thanks. You've answered my question. I was bothered by some weird text from the images.

**Reply by JorjMcKie** [reply]
Ok, good! That was easy then 😎

**Reply by yusheng0104** [reply]
Could also exclude the metadata of the images? In the documentation, it says TEXT_PRESERVE_IMAGES will return metadata.
Thanks,

**Comment by JorjMcKie** [comment]
If bit ``TEXT_PRESERVE_IMAGES`` is switched off, no image information at all will be included in any output type. Also no meta information.

**Reply by yusheng0104** [reply]
Thanks a lot. Could you please teach me how to switch TEXT_PRESERVE_IMAGES off?

**Reply by JorjMcKie** [reply]
Use the Python "OR" operator ``"|"`` to set single bits of the ``flags`` argument.
Depending on the text method used, there are default combinations used, if you do **_not specify_** flags - see [here](https://pymupdf.readthedocs.io/en/latest/app2.html#text-extraction-flags-defaults).
Otherwise use [this](https://pymupdf.readthedocs.io/en/latest/vars.html#text-extraction-flags) list to determine the bits you need:
```python
flags = (fitz.TEXT_PRESERVE_LIGATURES |
    fitz.TEXT_PRESERVE_WHITESPACE
    )
```
Then use ``flags`` in your text extraction or search function. The above example will ignore all images.

**Reply by yusheng0104** [reply]
Thanks. I tried _get_text(options = "text", flags =3)_, but still got text from the images.

**Reply by JorjMcKie** [reply]
I am sure there is some misconception: this extraction format does never access image information!
Can you give me the file?
If you have confidentiality concerns, you may use my private e-mail jorj.x.mckie@outlook.de.
